### PR TITLE
cartridges: init at 2.0.4

### PR DESCRIPTION
**Pull request by getchoo**
###### Description of changes

[cartridges](https://github.com/kra-mo/cartridges) is a simple game launcher written in python using libadwaita. it supports managing games from platforms/programs such as steam, lutris, heroic, bottles, and itch.io

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/cartridges/default.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        It has support for importing games from Steam, Lutris, Heroic
+        and more with no login necessary.
+        You can sort and hide games or download cover art from SteamGridDB.
+      '';
+      homepage = "https://apps.gnome.org/app/hu.kramo.Cartridges/";
+      license = licenses.gpl3Plus;
+      maintainers = [maintainers.getchoo];
+      platforms = platforms.linux;
+    };
+  }
```
Comment (BenediktBroich) [inline pkgs/applications/misc/cartridges/default.nix:77]:
```suggestion
{ blueprint-compiler
, desktop-file-utils
, fetchFromGitHub
, fetchFromGitLab
, gobject-introspection
, lib
, libadwaita
, meson
, ninja
, python3
, stdenv
, testers
, wrapGAppsHook4
}:
stdenv.mkDerivation (finalAttrs: {
  pname = "cartridges";
  version = "1.5.4";

  src = fetchFromGitHub {
    owner = "kra-mo";
    repo = "cartridges";
    rev = "v${finalAttrs.version}";
    sha256 = "sha256-pOGsuhsBZTFcq+1xsHXtBabV8i9pQw3w6UJnVE3EihM=";
  };

  buildInputs = [
    gobject-introspection
    libadwaita
    (python3.withPackages (p: with p; [
      pillow
      pygobject3
      pyyaml
      requests
    ]))
  ];

  nativeBuildInputs = [
    blueprint-compiler
    desktop-file-utils
    meson
    ninja
    wrapGAppsHook4
  ];

  passthru.tests.version = testers.testVersion {
    package = finalAttrs.finalPackage;
  };

  meta = with lib; {
    description = "A GTK4 + Libadwaita game launcher";
    longDescription = ''
      A simple game launcher for all of your games.
      It has support for importing games from Steam, Lutris, Heroic
      and more with no login necessary.
      You can sort and hide games or download cover art from SteamGridDB.
    '';
    homepage = "https://apps.gnome.org/app/hu.kramo.Cartridges";
    license = licenses.gpl3Plus;
    maintainers = with maintainers; [ benediktbroich getchoo ];
    platforms = platforms.unix;
  };
})
```

File: pkgs/applications/misc/cartridges/default.nix [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  libadwaita,
+  meson,
+  ninja,
+  python3,
+  stdenv,
+  wrapGAppsHook4,
+  ...
+}: let
+  inherit (lib) licenses maintainers platforms;
+
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/cartridges/default.nix:17]:
```suggestion
```

File: pkgs/applications/misc/cartridges/default.nix [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    nativeBuildInputs = [
+      blueprint-compiler'
+      desktop-file-utils
+      meson
+      ninja
+      wrapGAppsHook4
+    ];
+
+    meta = {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/cartridges/default.nix:64]:
```suggestion
    meta = with lib; {
```

File: pkgs/applications/misc/cartridges/default.nix [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  fetchFromGitLab,
+  gobject-introspection,
+  lib,
+  libadwaita,
+  meson,
+  ninja,
+  python3,
+  stdenv,
+  wrapGAppsHook4,
+  ...
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/cartridges/default.nix:14]:
```suggestion
  wrapGAppsHook4
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/cartridges/default.nix (hunk 1 above)
Comment (getchoo) [reply pkgs/applications/misc/cartridges/default.nix]: i took most of the suggestions here besides using passthru for the version. i just prefer `rec`

**Review by BenediktBroich — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/cartridges/default.nix [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      domain = "gitlab.gnome.org";
+      owner = "jwestman";
+      repo = "blueprint-compiler";
+      rev = "v${version}";
+      hash = "sha256-3lj9BMN5aNujbhhZjObdTOCQfH5ERQCgGqIAw5eZIQc=";
+    };
+
+    doCheck = false;
+  });
+in
```
[the remark this answers is not in the document]
Comment (getchoo) [reply pkgs/applications/misc/cartridges/default.nix:29]: oops! i must have been looking at 23.05 😆

**Review by BenediktBroich — APPROVED** [review]

**Comment by BenediktBroich** [comment]
@ofborg build cartridges

File: pkgs/applications/misc/cartridges/default.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "cartridges";
+  version = "2.0.1";
+
+  src = fetchFromGitHub {
+    owner = "kra-mo";
+    repo = "cartridges";
+    rev = "v${finalAttrs.version}";
+    sha256 = "sha256-ApPO8OhKVg4Wcp+oYHLcgLksi0muOpGlv/p0zsKxTM4=";
```
Comment (BenediktBroich) [inline pkgs/applications/misc/cartridges/default.nix:22]:
```suggestion
    sha256 = "sha256-vtfSDyJeqsyMBkDIqcXceYZi9ODcB/YinLg5qp606zk=";
```

**Review by BenediktBroich — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/cartridges/default.nix (hunk 6 above)
Comment (getchoo) [reply pkgs/applications/misc/cartridges/default.nix:22]: thanks for catching this! i was using `nurl` and forgot to put a space between the repo url and the tag :/

File: pkgs/applications/misc/cartridges/default.nix [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    blueprint-compiler
+    desktop-file-utils
+    gobject-introspection
+    meson
+    ninja
+    wrapGAppsHook4
+  ];
+
+  passthru.tests.version = testers.testVersion {
+    package = finalAttrs.finalPackage;
```
[the remark this answers is not in the document]
Comment (BenediktBroich) [reply pkgs/applications/misc/cartridges/default.nix:46]: Had the same Issues when packaging blueprint-compiler. Look [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/blueprint/default.nix#L50C8-L50C8).

[the remark this answers is not in the document]
Comment (getchoo) [reply pkgs/applications/misc/cartridges/default.nix]: i've removed this test as i didn't realize cartridges doesn't have a `--version` argument to test.
```
$ nix run .#cartridges -- --version
Unknown option --version
```

[the remark this answers is not in the document]
Comment (BenediktBroich) [reply pkgs/applications/misc/cartridges/default.nix]: Does it have --help with the version number?

**Comment by BenediktBroich** [comment]
Result of `nixpkgs-review pr 234486` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cartridges</li>
  </ul>
</details>

**Review by BenediktBroich — APPROVED** [review]

**Pull request merged**